azerothcore/azerothcore-wotlk
# Fix (scripts/Scholomance) - Kirtonos the Herald event now starts correctly again

The root cause of the issue is that when a creature is summoned by an instance, the ``IsSummonedBy()`` method is not called anymore. 

Additionally, I noticed that he's swimming through the air, so I checked how it's done for Onyxia and tried to copy from there.

## Changes Proposed:
-  Use `DoAction` on the summoned creature (Kirtonos). Reuse the `IN_PROGRESS` define because it doens't hurt..
- Add NO_MOVE_FLAGS_UPDATE, I copied the query from Keira3, I don't recall how they're supposed to be made if it should OR or actually use absolute values.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14944

## SOURCE:


## Tests Performed:

- Ran it on my own computer, the boss seems to start moving and not swim in the air in humanoid form.
-  I did notice earlier on that he doesn't always start, so maybe we want to have a few tries to be sure.


## How to Test the Changes:


1. .teleport scholomance
2. Enter the dungeon
3. .go xyz 325.122009 93.6436000 101.637001
4. .add 13523
5. right click the brazier
6. Kirtonos should spawn and start moving (flying around) and then come to the place where you are. He should behave normally after that, no swimming in the air.

## Known Issues and TODO List:



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
